sass/components/_on-this-page.scss

```diff
@@ -0,0 +1,36 @@
+/// internal page navigation
+/// i.e., the "On This Page" section
+
+
+.tree-menu .on-this-page {
+    grid-row: 2;
+    padding-inline: 0.5em;
+    // border: 3px solid rgba($color-white, 0.1);
+    border: 3px solid transparent;
+    border-width-top: 0;
+    box-sizing: border-box;
+    background-color: $color-grey-800;
+}
+
+.page-with-menu .tree-menu .on-this-page {
+    // desktop menu uses grey-800 as the bg
+    background-color: $color-grey-900;
```

Comment:
I would add this as a "modifier", like `.on-this-page--mobile` and apply that class to the mobile menu. `docs_menu_page_row` macro can be refactored to accept an extra parameter `is_mobile: bool` and based on that you could apply `.on-this-page--mobile`.

If we add `is_mobile` to `docs_menu_page_row` then `prefix` parameter could be removed and computed internally based on `is_mobile`. So we would have a couple of calls like:

```
{{ docs_macros::docs_menu(is_mobile=true, root=get_section(path=root_section_path) ) }}
{{ docs_macros::docs_menu(is_mobile=false, root=get_section(path=root_section_path) ) }}
```

Reply:
changed it so that it just uses a translucent background so there's no distinction between the mobile and desktop version